### PR TITLE
backend: remove generated files and change to typescript

### PR DESCRIPTION
**Pull request by reginawang3495**
add node_modules and other generated files to .gitignore
cut out unnecessary dependencies that make it difficult to start backend up
cleanup README.md

### CODE REVIEW

**Review by ghuser1023 — COMMENTED** [review]
Are we sure we want to commit `node_modules`? I'm not up to date on the current status of this debate in the community.

**Comment by reginawang3495** [comment]
oh yeee nice catch, i removed

**Review by ghuser1023 — COMMENTED** [review]
Running `npm install` modifies a few files:
```
On branch backend
Your branch is up to date with 'origin/backend'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   package-lock.json
	modified:   yarn.lock

no changes added to commit (use "git add" and/or "git commit -a")
```
I'm using `npm 7.3.0` on macOS. Is this an issue?

**Comment by reginawang3495** [comment]
oh the lock files often change when npm install is run because of different npm versions and stuffs
npm is backwards compatible so it should be fine, but if you want, you can install the newest one with us: https://nodejs.org/en/ (v14)
tldr: naw, thats normal, we chillin :)

**Comment by ghuser1023** [comment]
`npm install` quite literally rewrote all the hashes in `yarn.lock`, though -- not sure that's desired behavior?

On that note, I checked the linked walkthrough and it seems to be using `yarn` instead of `npm`. Wiping `node_modules` and running `yarn install` didn't modify any lockfiles, but it did warn about the presence of two of them:
```
yarn install v1.22.10
warning package-lock.json found. Your project contains lock files generated by tools other than Yarn. It is advised not to mix package managers in order to avoid resolution inconsistencies caused by unsynchronized lock files. To clear this warning, remove package-lock.json.
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 🔨  Building fresh packages...
✨  Done in 4.07s.
```

Yeah, should have specified node version as well: I have `npm` and `node` managed by brew, so `node` is actually at version 15.6.0. I think (?) that's not the LTS version, so if that causes any issues I can probably manually install 14 instead.

**Comment by reginawang3495** [comment]
k i removed yarn.lock

**Review by ghuser1023 — APPROVED** [review]
`package-lock.json` doesn't seem like much of a lockfile, but I guess I'm okay with that 😅
